image/helpers.go

```diff
@@ -342,13 +347,96 @@ func (tsto *ToolingStore) DownloadSnap(name string, opts DownloadOptions) (targe
 	}()
 
 	dlOpts := &store.DownloadOptions{LeavePartialOnError: opts.LeavePartialOnError}
-	if err = sto.Download(context.TODO(), name, targetFn, &snap.DownloadInfo, pb, tsto.user, dlOpts); err != nil {
-		return "", nil, "", err
+	if err = tsto.sto.Download(context.TODO(), snap.SnapName(), targetFn, &snap.DownloadInfo, pb, tsto.user, dlOpts); err != nil {
+		return nil, err
 	}
 
 	signal.Reset(syscall.SIGINT)
 
-	return targetFn, snap, redirectChannel, nil
+	return &DownloadedSnap{
+		Path:            targetFn,
+		Info:            snap,
+		RedirectChannel: redirectChannel,
+	}, nil
+}
+
+type SnapDownloadOptions struct {
```

Comment:
or maybe this could be called `SnapDownloadRef`, `SnapDownloadRequest`?

Reply:
or SnapToDownload since this is passed as `toDownload` list